Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp

```diff
@@ -488,6 +530,37 @@ void AsyncScrollingCoordinator::animatedScrollDidEndForNode(ScrollingNodeID scro
     }
 }
 
+void AsyncScrollingCoordinator::wheelEventScrollWillStartForNode(ScrollingNodeID scrollingNodeID)
+{
+    ASSERT(isMainThread());
+
+    if (!m_page)
+        return;
+
+    auto* frameView = frameViewForScrollingNode(scrollingNodeID);
+    if (!frameView)
+        return;
+
+    m_scrollingHysterisisTimer.stop();
+#if USE(APPKIT)
+    m_page->editorClient().setCaretDecorationVisibility(false);
+#endif
+}
+
+void AsyncScrollingCoordinator::wheelEventScrollDidEndForNode(ScrollingNodeID scrollingNodeID)
+{
+    ASSERT(isMainThread());
+
+    if (!m_page)
+        return;
+
+    auto* frameView = frameViewForScrollingNode(scrollingNodeID);
+    if (!frameView)
+        return;
+
+    m_scrollingHysterisisTimer.startOneShot(200_ms);
```

Comment:
It's wrong for a timer owned by AsyncScrollingCoordinator to control caret behavior. Move this timer to some class that's related to the caret.